ocaml/idl/datamodel.ml

```diff
@@ -1691,6 +1691,61 @@ module Data_source = struct
       ()
 end
 
+module Sr_stat = struct
+  let health = Enum ("sr_health", [
+      "healthy", "Storage is fully available";
+      "recovering", "Storage is busy recovering, e.g. rebuilding mirrors.";
+    ])
+
+  let t =
+    let lifecycle = [Published, rel_kolkata, ""] in
+    create_obj
+      ~in_db:false
+      ~persist:PersistNothing
+      ~gen_constructor_destructor:false
+      ~lifecycle
+      ~in_oss_since:None
+      ~name:_sr_stat
+      ~descr:"A set of high-level properties associated with an SR."
+      ~gen_events:false
+      ~messages:[]
+      ~doccomments:[]
+      ~messages_default_allowed_roles:(Some []) (* No messages, so no roles allowed to use them *)
+      ~contents:
+        [ (* The uuid is not needed here and only adds inconvenience. *)
+          field ~qualifier:DynamicRO ~lifecycle ~ty:String "name_label" "Short, human-readable label for the SR.";
+          field ~qualifier:DynamicRO ~lifecycle ~ty:String "name_description" "Longer, human-readable description of the SR. Descriptions are generally only displayed by clients whe
+ the user is examining SRs in detail.";
+          field ~qualifier:DynamicRO ~lifecycle ~ty:Int "free_space" "Number of bytes free on the backing storage (in bytes)";
+          field ~qualifier:DynamicRO ~lifecycle ~ty:Int "total_space" "Total physical size of the backing storage (in bytes)";
+          field ~qualifier:DynamicRO ~lifecycle ~ty:Bool "clustered" "Indicates whether the SR uses clustered local storage.";
+          field ~qualifier:DynamicRO ~lifecycle ~ty:health "health" "The health status of the SR.";
```

Comment:
Here & in xcp-idl we return less info than what is provided by SMAPIv3 - maybe we could extend them?

Reply:
which info is missing?

Reply:
UUID, `sr` URI, and datasources: https://xapi-project.github.io/xapi-storage/volume.html#a-7